painting/src/main/java/com/ataulm/artcollector/painting/PaintingModule.kt

```diff
@@ -0,0 +1,25 @@
+package com.ataulm.artcollector.painting
+
+import android.arch.lifecycle.ViewModelProviders
+import com.ataulm.artcollector.painting.data.AndroidPaintingRepository
+import com.ataulm.artcollector.painting.domain.PaintingRepository
+import com.ataulm.artcollector.painting.ui.PaintingActivity
+import com.ataulm.artcollector.painting.ui.PaintingViewModel
+import com.ataulm.artcollector.painting.ui.PaintingViewModelFactory
+import dagger.Module
+import dagger.Provides
+
+@Module
+internal object PaintingModule {
+
+    @JvmStatic
+    @Provides
+    fun paintingRepository(paintingRepository: AndroidPaintingRepository): PaintingRepository {
```

Comment:
dagger knows how to instantiate AndroidPaintingRepository and if anything _depended_ on that concrete class, we wouldn't need this `provides`. but since things depend on the interface, we tell dagger that this is how it should handle requests for `PaintingRepository` (return an AndroidPaintingRepository)